app/src/main/java/com/conupods/OutdoorMaps/BuildingOverlays.java

```diff
@@ -49,39 +50,33 @@ public void removePolygons() {
         }
     }
 
-    private void addColorsToMarkers(GeoJsonLayer layer) {
-        // Iterate over all the features stored in the layer
-        for (GeoJsonFeature feature : layer.getFeatures()) {
-            // Check if the  property exists
-            GeoJsonPolygonStyle buildingStyle = layer.getDefaultPolygonStyle();
-            buildingStyle.setFillColor(0x80eac700);
-            buildingStyle.setStrokeColor(0x80555555);
-            buildingStyle.setStrokeWidth(5);
-        }
+    private void styleBuildingOverlays(GeoJsonLayer layer) {
+
+        GeoJsonPolygonStyle buildingStyle = layer.getDefaultPolygonStyle();
+        buildingStyle.setFillColor(0x80eac700);
+        buildingStyle.setStrokeColor(0x80555555);
+        buildingStyle.setStrokeWidth(5);
+
     }
 
     private class DownloadGeoJsonFile implements Runnable {
 
         @Override
         public void run() {
-            try {
-                // Open a stream from the URL
-                InputStream stream = new URL(mGeoStringLink).openStream();
+            // Open a stream from the URL
+            try (InputStream stream = new URL(mGeoStringLink).openStream();
+                 BufferedReader reader = new BufferedReader(new InputStreamReader(stream))) {
```

Comment:
I dont think you are still closing stream., shouldn't you add a finally block to close them

Reply:
try-with-resources will actually always close the streams. No need to explicitly call close in a finally block